### ISSUE TITLE
Woocommerce - no "Add to cart" button with variable products

### ISSUE THREAD

**Issue by dazzyon** [gantry/gantry5#853]
On the Woocommerce product detail page:

When i switch to the default theme (Twenty Fifteen) the button is visible. Also when i set the product to "simple product" the button works too.
When i enable the Hydrogen theme and set the product to "variable product" the "Add to cart" button disappears.

**Comment by newkind** [comment]
Do you use any 3rd party plugins ? Any link to your site ? What are the steps to replicate this ?

**Comment by dazzyon** [comment]
Disabling 3rd party plugins doesn't solve the problem. Also the problem doesn't show with the default Wordpress theme.
The website is not online yet unfrotunately.

The steps to replicate (after installing Wordpress, Timber, Gantry5 and Woocommerce):

- Add an "attibute" to the Woocommerce products (for example colour)
- Add a "variable product" to the Woocommerce shop
- Add the created attribute to the product and set the values (for example "red" and "blue") and select "used for variations"
- Add the variations to the product (one for each value of the colour attribute)
- Set the price and sku for each variation

Now you have a product in the shop, with the colour variations "red" and "blue". With the default theme the "Add to cart" button is visible on the product detail page. The button disappears when using the Gantry5 (Hydrogen) theme.

**Comment by newkind** [comment]
I was just investigating it. The issue is that when WooCommerce tries to enqueue wc-add-to-cart-variation script, it's not even registered yet, thus there's nothing to enqueue. I'll try to check if there can be something done about it without deregistering and registering the script again.

**Comment by newkind** [comment]
I know what's the issue, but I'm not sure if we'll be able to change this behavior in the core.
In the meantime you can add this to your `functions.php` to make it work :
```
add_action( 'wp_enqueue_scripts', 'wc_cart_variation_script' );
function wc_cart_variation_script() {
    global $woocommerce;

    if( defined( 'WOOCOMMERCE_VERSION' ) && is_woocommerce() ) {
        if( is_single() && get_post_type() == 'product' ) {
            wp_enqueue_script( 'wc-add-to-cart-variation', $woocommerce->plugin_url() . '/assets/js/frontend/add-to-cart-variation.js', array('jquery'), '1.6', true );
        }
    }
}
```

**Comment by dazzyon** [comment]
Thanks for the manual solution, i can confirm this works!

**Comment by newkind** [comment]
No problem, we'll try to see if this can be fixed in the core ;)

**Comment by mahagr** [comment]
Fixed in #855

**Comment by newkind** [comment]
With the next Gantry update this should be fixed, but you'll have to update your php files in theme root directory.
